File: backend/albergue1/services/mascota.py
from flask import Blueprint, request, jsonify, make_response
from utils.db import db

from models.mascota import Mascota
from schemas.mascota import mascota_schema, mascotas_schema

mascota_routes = Blueprint('mascota_routes', __name__)

@mascota_routes.route("/get_mascotas", methods=["GET"])
def get_mascotas():
    mascotas = Mascota.query.all()

    if not mascotas:
        data = {
            'message': 'No se encontraron mascotas.',
            'status': 404
        }
        return make_response(jsonify(data), 404)
    else:
        result = mascotas_schema.dump(mascotas)
        data = {
            'message': 'Mascotas encontradas.',
            'status': 200,
            'data': result
        }
        return make_response(jsonify(data), 200)
    
@mascota_routes.route("/get_mascota", methods=["POST"])
def get_mascota():
    id_mascota = request.json.get('id_mascota')
    if not id_mascota:
        data = {
            'message': 'ID de mascota no proporcionado.',
            'status': 400
        }
        return make_response(jsonify(data), 400)
    else:
        mascota = Mascota.query.get(id_mascota)

        if not mascota:
            data = {
                'message': 'Mascota no encontrada.',
                'status': 404
            }
            return make_response(jsonify(data), 404)
        else:
            result = mascota_schema.dump(mascota)
            data = {
                'message': 'Mascota encontrada.',
                'status': 200,
                'data': result
            }
            return make_response(jsonify(data), 200)

File: backend/albergue3/controllers/mascota.js
const Mascota = require('../models/mascota.js');

const get_mascotas = (req, res) => {
    Mascota.obtenerTodos((err, resultados) => {
        if (err) {
            return res.status(500).json({ mensaje: 'Error en el servidor', error: err });
        }
        if (resultados.length === 0) {
            return res.status(404).json({ mensaje: 'No se encontraron mascotas' });
        }
        res.status(200).json({
            message: 'Mascotas encontradas.',
            status: 200,
            data: resultados
        });
    });
};

const get_mascota = (req, res) => {
    const { id_mascota } = req.body;
    if (!id_mascota) {
        return res.status(400).json({ mensaje: 'ID de mascota es requerido' });
    }
    
    Mascota.obtenerPorId(id_mascota, (err, resultado) => {
        if (err) {
            return res.status(500).json({ mensaje: 'Error en el servidor', error: err });
        }
        if (!resultado || resultado.length === 0) {
            return res.status(404).json({ mensaje: 'Mascota no encontrada' });
        }
        res.status(200).json({
            message: 'Mascota encontrada.',
            status: 200,
            data: resultado[0]
        });
    });
}

module.exports = {
    get_mascotas,
    get_mascota
};
